File: puzzle2/puzzle2a.nim
import std/strformat, strutils

#..............................................................................
proc runSubmarine() =
    var position = 0
    var depth = 0

    let f = open("./input.txt")
    defer: f.close()

    var line: string

    while f.readLine(line):
        var s = line.strip().split(' ')
        var direction = s[0]
        var distance = parseInt(s[1])
        echo(&"{direction} {distance}")

        case direction:
            of "forward":
                position += distance
            of "up":
                depth -= distance

                if depth < 0:
                    echo("BREACH!!")
            of "down":
                depth += distance

    var hp = position * depth
    echo(&"position: {position}, depth: {depth}. total: {hp}")

#..............................................................................
runSubmarine()

File: day 02/puzzle2a.py
# Open and parse the input file
# Parse the input file into a list of instructions separated by space

position = 0
depth = 0
aim = 0

with open('./input.txt') as f:
    for line in f:
        direction, distance = line.split()
        distance = int(distance)
        print(f'{direction} {distance}')

        match direction:
            case 'forward':
                position += distance
                depth += aim * distance
            case 'up':
                # depth -= distance
                aim -= distance

                if depth < 0:
                    print("BREACH!!")
            case 'down':
                # depth += distance
                aim += distance

    hp = position * depth
    print(f'position: {position}, depth: {depth}, aim: {aim}. total: {hp}')
